### ISSUE TITLE
Prebid custom key bug fix

### ISSUE THREAD

**Issue by patrickloughrey** [prebid/Prebid.js#6734]
## Type of issue
Bug Fix

## Description
I have noticed a potential Prebid core bug when troubleshooting one of our publishers' pages. I am implementing Prebid custom keys via _pbjs.bidderSettings_ defined here: https://docs.prebid.org/dev-docs/publisher-api-reference.html#module_pbjs.bidderSettings

The issue that I am seeing is that when explicitly defining these custom keys via _pbjs.bidderSettings.adServerTargeting_ I am seeing both the custom keys and the Prebid default keys being sent to Google Ad Manager in the ad server request. The Prebid default keys should not be sent since the _sendStandardTargeting_ flag is set to _false_. Here is a screenshot of the ad server request within a test page environment:
<img width="1334" alt="Screen Shot 2021-05-10 at 3 16 21 PM" src="https://user-images.githubusercontent.com/20526608/117722214-ca379c00-b1ae-11eb-8663-4a225c4e186a.png">

The code used to implement the custom keys is the following:
```
pbjs.bidderSettings = {
    improvedigital: {
         sendStandardTargeting: false,
         adserverTargeting: [
            {
               key: "hb_pb_id",
               val: function(bidResponse) {
                   return bidResponse.pbMg;
               }
            }, {
               key: "hb_size_id",
               val: function(bidResponse) {
                   return bidResponse.size;
               }
            }, {
               key: "hb_adid_id",
               val: function(bidResponse) {
                   return bidResponse.adId;
               }
            }, {
               key: "hb_uuid_id",
               val: function(bidResponse) {
                   return bidResponse.videoCacheKey;
               }
            }
         ]
     },
     rubicon: {
         sendStandardTargeting: false,
         adserverTargeting: [
            {
               key: "hb_pb_rubi",
               val: function(bidResponse) {
                   return bidResponse.pbMg;
               }
            }, {
               key: "hb_size_rubi",
               val: function(bidResponse) {
                   return bidResponse.size;
               }
            }, {
               key: "hb_adid_rubi",
               val: function(bidResponse) {
                   return bidResponse.adId;
               }
            }, {
               key: "hb_uuid_rubi",
               val: function(bidResponse) {
                   return bidResponse.videoCacheKey;
               }
            }
         ]
     },
    '33across': {
         sendStandardTargeting: false,
         adserverTargeting: [
            {
               key: "hb_pb_33a",
               val: function(bidResponse) {
                   return bidResponse.pbMg;
               }
            }, {
               key: "hb_size_33a",
               val: function(bidResponse) {
                   return bidResponse.size;
               }
            }, {
               key: "hb_adid_33a",
               val: function(bidResponse) {
                   return bidResponse.adId;
               }
            }, {
               key: "hb_uuid_33a",
               val: function(bidResponse) {
                   return bidResponse.videoCacheKey;
               }
            }
         ]
     }
}
```

I have verified that _pbjs.bidderSettings_ is successfully updated and reflected on the page with the _sendStandardTargeting_ flag set to _false_:
<img width="779" alt="Screen Shot 2021-05-10 at 3 37 44 PM" src="https://user-images.githubusercontent.com/20526608/117721133-7a0c0a00-b1ad-11eb-8bcc-63348a35577b.png">

## Steps to reproduce
This bug can be reproduced using the following test page:

[prebid-bug.zip](https://github.com/prebid/Prebid.js/files/6454933/prebid-bug.zip)

You can serve this test page up with any server. In order to see Prebid key values sent to the ad server, a bid spoofing mechanism will have to be used in conjunction.

**Comment by dgirardi** [comment]
The issue is that "landscape" targeting keys (those suffixed with `_<bidder>` generated with `enableSendAllTargeting`) do not look at `bidderSettings`. It's not right, but it's also not immediately clear how they should interact. 

https://github.com/prebid/Prebid.js/blob/c86d069c3b5ba7867d49abdb9787b4af463ffddf/src/targeting.js#L597-L618

**Comment by dgirardi** [comment]
@bretg this is an example of a "breaking fix": in https://github.com/prebid/Prebid.js/issues/8130 the proposal is to do as this issue suggests, remove standard bidder KVPs if `sendStandardTargeting === false`. But we may have publishers that are relying on the current behavior.

Should we introduce a new setting as a stopgap (maybe `sendAllBidsStandardTargeting`), or wait until we redo KVPs?

**Comment by bretg** [comment]
This was originally opened by a Magniter, but I haven't heard internal conversation calling out this as a high priority issue. 

So unless someone else disagrees, I'm ok tying this to the KVP control refactoring planned in https://github.com/prebid/Prebid.js/issues/8130

**Comment by bretg** [comment]
closing this -- will be resolved with #8130
